x-pack/plugins/observability_solution/inventory/public/analytics/register_eem_enabled_context.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AnalyticsServiceSetup } from '@kbn/core/public';
+import { BehaviorSubject } from 'rxjs';
+import { EntityManagerPublicPluginSetup } from '@kbn/entityManager-plugin/public';
+
+export let eemEnabled$: BehaviorSubject<{ eem_enabled: boolean }>;
+
+export function registerEemEnabledContext(
+  analytics: AnalyticsServiceSetup,
+  entityManager: EntityManagerPublicPluginSetup
+) {
+  entityManager.entityClient.isManagedEntityDiscoveryEnabled().then(({ enabled }) => {
```

Comment:
I’m considering using a context for the EEM state. 

I’ll initialize it during the plugin setup and use it where we need it, mainly where we are using `useEntityManager`.

I'm not 100% sure if it's a possible option since I'm still getting familiar with how React works, but I think it could be a good approach.